### PR TITLE
Fix color serialization and deserialization

### PR DESCRIPTION
**Pull request by z3dev**
These changes correct the serialize and deserialize of geometries.

In general, colors are serialized using:
- color on geometry converted to color on shape (external format)
- color on polygon converted to color on face (external format)

In general, colors are deserialized using:
- color on shape to color on geometry
- color on face to color on polygon

There are related fixes to modeling, as the new modifiers (snap and triangulate) were not maintaining the colors assigned to polygons.

There is also a fix to the CLI, to allow serialization parameters to be passed through. This allows serialization and deserialization options to be specified on the command line. Such as:
- target geometry
- units
- segments

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Does your submission pass tests?

Additional testing was conducted with deserialization of all sample files to geometry.
Additional testing was conducted with 2D and 3D geometries to shapes (all available external formats)

### CODE REVIEW

**Review by z3dev — COMMENTED** [review]
Changes as expected.

File: packages/cli/src/generateOutputData.js [hunk 1]
```diff
@@ -47,7 +47,8 @@ const generateOutputData = (source, params, options) => {
 
     // convert any inputs
     const prevsource = source
-    source = conversionTable[inputFormat]({ source, params, options })
+    const deserializerOptions = Object.assign({}, params, options)
+    source = conversionTable[inputFormat]({ source, params, options: deserializerOptions })
```
Comment (mister-jscad) [inline packages/cli/src/generateOutputData.js:51]: do deserializers have an 'params' option?

File: packages/io/amf-serializer/README.md [hunk 2]
```diff
@@ -23,7 +23,9 @@ This serializer outputs a 'blobable' array of data from one or more JSCAD geomet
 The array of data can either be used to create a Blob (`new Blob(blobable)`), or converted to a Node.js buffer.
 
 The serialization of the following geometries are possible.
-- serialization of 3D geometry (geom3) to AMF geometry (non-overlapping volumes)
+- serialization of 3D geometry (geom3) to AMF object (a unique mesh containing both vertices and volumes)
```
Comment (mister-jscad) [inline packages/io/amf-serializer/README.md:26]: These comments in the jsdoc would be nice as well.

**Review by mister-jscad — CHANGES_REQUESTED** [review]
Food for thoughts... and changes.

**Review by mister-jscad — APPROVED** [review]
Changes look good.

**Pull request merged**